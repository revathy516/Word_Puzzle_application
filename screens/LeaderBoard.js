import React, { useState, useEffect } from 'react';
import {
  StyleSheet,
  SafeAreaView,
  Share, View, Button
} from "react-native"
import Leaderboard from 'react-native-leaderboard';
import {readScore} from '../assets/scoreStorage';
import AsyncStorage from '@react-native-async-storage/async-storage';

const LeaderBoard = (props) => {
  // Hardcoded data, since in reality it will be available in cloud and consumed by webservice call.
  const datas = 
     [
        {userName:"revathy", highScore: "10"},
        {userName: "Mark", highScore: "5"},
        {userName: "kenny", highScore: "5"},
        {userName: "Gowtham", highScore: "5"}

        //...
    ]
    const onShare = async () => {
      try {
        const result = await Share.share({
          message:
            'join Word Puzzle and win high score '
        });
        if (result.action === Share.sharedAction) {
          if (result.activityType) {
            // shared with activity type of result.activityType
          } else {
            // shared
          }
        } else if (result.action === Share.dismissedAction) {
          // dismissed
        }
      } catch (error) {
        Alert.alert(error.message);
      }
    };

  return (
    <SafeAreaView style={styles.container}>
      <Leaderboard 
        data={datas} 
        sortBy='highScore' 
        labelBy='userName'/>
         <View style= {styles.categoryButton}>
         <Button  onPress={onShare} title="Share"  />
                        </View>
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({

  container: {
    justifyContent: "space-between",
    flex: 1,
  },
  categoryButton:{
    marginVertical:10,
     marginHorizontal:20
  },
})
export default LeaderBoard;

