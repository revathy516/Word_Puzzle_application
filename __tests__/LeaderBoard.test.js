import React from 'react';
import renderer from 'react-test-renderer';
import LeaderBoard from '../screens/LeaderBoard';

it('renders correctly', () => {
    const tree = renderer.create(<LeaderBoard />).toJSON();
    expect(tree).toMatchSnapshot();
  });