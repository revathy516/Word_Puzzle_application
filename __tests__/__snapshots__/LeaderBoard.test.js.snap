// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "highScore": 10,
          "userName": "Game 2",
        },
        Object {
          "highScore": 5,
          "userName": "Game 1",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#d6d7da",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              },
              Object {
                "backgroundColor": "#f2f5f7",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "marginRight": 5,
                  },
                  undefined,
                  Object {
                    "paddingLeft": 16,
                    "paddingRight": 6,
                  },
                ]
              }
            >
              1
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 17,
                    "paddingRight": 80,
                  },
                  undefined,
                ]
              }
            >
              Game 2
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "paddingLeft": 15,
                  "position": "absolute",
                  "right": 15,
                },
                undefined,
              ]
            }
          >
            10
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#d6d7da",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 15,
                "paddingTop": 15,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "marginRight": 5,
                  },
                  undefined,
                  Object {
                    "paddingLeft": 16,
                    "paddingRight": 6,
                  },
                ]
              }
            >
              2
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 17,
                    "paddingRight": 80,
                  },
                  undefined,
                ]
              }
            >
              Game 1
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "paddingLeft": 15,
                  "position": "absolute",
                  "right": 15,
                },
                undefined,
              ]
            }
          >
            5
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Share
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
